drivers/rail/rail_netdev.c

```diff
@@ -0,0 +1,496 @@
+            break;
+        case (NETOPT_AUTOACK):
+            if (RAIL_IsAutoAckEnabled(dev->rhandle) == true) {
+                *((netopt_enable_t *)val) =  NETOPT_ENABLE;
+            }
+            else {
+                *((netopt_enable_t *)val) =  NETOPT_DISABLE;
+            }
+            ret = sizeof(netopt_enable_t);
+            break;
+        case (NETOPT_PRELOADING):   // todo what does it do??
+            break;
+        case (NETOPT_RAWMODE):
+            break;
+        case (NETOPT_CSMA):         // todo can it be switched on / off?
+            *((netopt_enable_t *)val) = NETOPT_ENABLE;
+            ret = sizeof(netopt_enable_t);
+            break;
+        case (NETOPT_CSMA_RETRIES):
+            assert(max_len >= sizeof(int8_t));
+            *((uint8_t *)val) = csma_config.csmaTries;
+            ret = sizeof(uint8_t);
+            break;
+        case (NETOPT_AUTOCCA):
+            break;
+        case (NETOPT_MAC_NO_SLEEP):
+            if (dev->macNoSleep == false) {
+                *((netopt_enable_t *)val) =  NETOPT_DISABLE;
+            }
+            else {
+                *((netopt_enable_t *)val) =  NETOPT_ENABLE;
+            }
+            ret = sizeof(netopt_enable_t);
+            break;
+        case (NETOPT_CHANNEL_HOP):
+            break;
+        case (NETOPT_CODING_RATE):
+            break;
+
+        case (NETOPT_BANDWIDTH):
+            /*
+               assert(max_len >= sizeof(int8_t));
+               // bits/seconds
+               uint32_t bw = RAIL_BitRateGet();
+               uint8_t bw_kb = bw/1000;
+             *((uint8_t *)val) = bw_kb;
+               ret = sizeof(uint8_t);
+             */
+            break;
+        case (NETOPT_CHANNEL_FREQUENCY):
+            break;
+        case (NETOPT_SINGLE_RECEIVE):
+            break;
+        case (NETOPT_IQ_INVERT):
+            break;
+        case (NETOPT_SRC_LEN):
+            break;
+        case (NETOPT_ADDRESS_LONG):
+            break;
+        case (NETOPT_DEVICE_TYPE):
+            break;
+        case (NETOPT_ADDRESS):
+            break;
+        case (NETOPT_CHANNEL):
+            break;
+        case (NETOPT_NID):
+            break;
+        case (NETOPT_SPREADING_FACTOR):
+            break;
+        case (NETOPT_ACK_REQ):
+            break;
+        case (NETOPT_STATS):
+            break;
+        default:
+            DEBUG("not supported netopt code at rail drv %d str %s \n", opt, netopt2str(opt));
+            break;
+    }
+    if (ret != -ENOTSUP) {
+        return ret;
+    }
+    // delegate cases to ieee802.15.4 layer
+    //NETOPT_PROTO
+    //
+
+    ret = netdev_ieee802154_get((netdev_ieee802154_t *)netdev, opt, val, max_len);
+
+    if (ret != -ENOTSUP) {
+        return ret;
+    }
+
+    //   DEBUG("ieee802.15.4 could not handle netopt opt %s \n", netopt2str(opt));
+
+    return ret;
+}
+
+static int _set(netdev_t *netdev, netopt_t opt, const void *val, size_t len)
+{
+
+    if (netdev == NULL) {
```

Comment:
Guess that [`netdev_ieee802154_set`](https://github.com/RIOT-OS/RIOT/blob/8fd85e1c133715a00baddf65b9f638c341e3fc9e/drivers/netdev_ieee802154/netdev_ieee802154.c#L145) can be used here?